tools/setup_env.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+# This script sets up the environment for the Ziggurat test suite.
+
+# The subnet to add to loopback devices
+SUBNET="1.1.1.0/24"
+
+# Rippled files
+if [ -z "$RIPPLED_BIN_PATH" ]; then
+    echo "Aborting. Export RIPPLED_BIN_PATH before running this script."
+    exit 1
+fi
+RIPPLED_BIN_NAME="rippled"
+
+# Ziggurat config files
+ZIGGURAT_RIPPLED_DIR="$HOME/.ziggurat/ripple"
+ZIGGURAT_RIPPLED_SETUP_DIR="$ZIGGURAT_RIPPLED_DIR/setup"
+ZIGGURAT_RIPPLED_SETUP_CFG_FILE="$ZIGGURAT_RIPPLED_SETUP_DIR/config.toml"
+ZIGGURAT_RIPPLED_TESTNET_DIR="$ZIGGURAT_RIPPLED_DIR/testnet"
+ZIGGURAT_RIPPLED_STATEFUL_DIR="$ZIGGURAT_RIPPLED_DIR/stateful"
+
+setup_config_file() {
+    echo "--- Setting up configuration file"
+    echo "Creating $ZIGGURAT_RIPPLED_SETUP_CFG_FILE with contents:"
+    mkdir -p $ZIGGURAT_RIPPLED_SETUP_DIR
+    echo
+    echo "# Rippled installation path" > $ZIGGURAT_RIPPLED_SETUP_CFG_FILE
+    echo "path = \"$RIPPLED_BIN_PATH\"" >> $ZIGGURAT_RIPPLED_SETUP_CFG_FILE
+    echo "# Start command with possible arguments" >> $ZIGGURAT_RIPPLED_SETUP_CFG_FILE
+    echo "start_command = \"./$RIPPLED_BIN_NAME --silent\"" >> $ZIGGURAT_RIPPLED_SETUP_CFG_FILE
+
+    # Print file contents so the user can check whether the path is correct
+    cat $ZIGGURAT_RIPPLED_SETUP_CFG_FILE
+    echo
+}
+
+setup_initial_node_state() {
+    # Query only after a long delay to account for compilation times and network preparation work
+    ACCOUNT_QUERY_DELAY_MIN=5m
+
+    echo "--- Setting up initial node state, takes at least 5 minutes"
+    echo
+    echo "Spinning up a node instance, please be patient"
+    cargo t setup::testnet::test::run_testnet -- --ignored &
+    echo
+    sleep $ACCOUNT_QUERY_DELAY_MIN
+    echo "--- Querying account info"
+    # Run account query until it responds with "ResponseStatus.SUCCESS" or MAX_ATTEMPTS is reached
+    ACCOUNT_QUERY_TIMEOUT_SEC=5s
+    ACCOUNT_QUERY_MAX_ATTEMPTS=5
+    ACCOUNT_QUERY_NUM_ATTEMPTS=0
+    until [ $ACCOUNT_QUERY_NUM_ATTEMPTS -gt $(($ACCOUNT_QUERY_MAX_ATTEMPTS-1)) ] \
+        || timeout $ACCOUNT_QUERY_TIMEOUT_SEC python3 tools/account_info.py | grep "ResponseStatus.SUCCESS"; do
```

Comment:
for Mac we should use the `gtimeout` cmd here

Reply:
Done :white_check_mark: